tests/cover/test_datetimes.py

```diff
@@ -0,0 +1,165 @@
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import datetime as dt
+
+from flaky import flaky
+
+from hypothesis import find, given, settings
+from tests.common.debug import minimal
+from hypothesis.strategies import none, dates, times, binary, datetimes, \
+    timedeltas
+from hypothesis.strategytests import strategy_test_suite
+from hypothesis.internal.compat import hrange
+from hypothesis.searchstrategy.datetime import DatetimeStrategy
+from hypothesis.internal.conjecture.data import ConjectureData
+
+TestStandardDescriptorFeatures_timedeltas1 = strategy_test_suite(timedeltas())
+
+
+def test_can_find_positive_delta():
+    assert minimal(timedeltas(), lambda x: x.days > 0) == dt.timedelta(1)
+
+
+def test_can_find_negative_delta():
+    assert minimal(timedeltas(max_delta=dt.timedelta(10**6)),
+                   lambda x: x.days < 0) == dt.timedelta(-1)
+
+
+def test_can_find_on_the_second():
+    minimal(timedeltas(), lambda x: x.seconds == 0)
+
+
+def test_can_find_off_the_second():
+    minimal(timedeltas(), lambda x: x.seconds != 0)
+
+
+def test_simplifies_towards_zero_delta():
+    d = minimal(timedeltas())
+    assert d.days == d.seconds == d.microseconds == 0
+
+
+def test_min_value_is_respected():
+    assert minimal(timedeltas(min_delta=dt.timedelta(days=10))).days == 10
+
+
+def test_max_value_is_respected():
+    assert minimal(timedeltas(max_delta=dt.timedelta(days=-10))).days == -10
+
+
+@given(timedeltas())
+def test_single_timedelta(val):
+    assert timedeltas(val, val).example() is val
+
+
+TestStandardDescriptorFeatures_datetimes1 = strategy_test_suite(datetimes())
+
+
+def test_simplifies_towards_millenium():
+    d = minimal(datetimes())
+    assert d.year == 2000
+    assert d.month == d.day == 1
+    assert d.hour == d.minute == d.second == d.microsecond == 0
+
+
+@given(datetimes())
+def test_default_datetimes_are_naive(dt):
+    assert dt.tzinfo is None
+
+
+@flaky(max_runs=3, min_passes=1)
+def test_bordering_on_a_leap_year():
+    with settings(database=None, max_examples=10 ** 7, timeout=-1):
+        x = minimal(datetimes(dt.datetime.min.replace(year=2003),
+                              dt.datetime.max.replace(year=2005)),
+                    lambda x: x.month == 2 and x.day == 29)
+    assert x.year == 2004
+
+
+def test_DatetimeStrategy_draw_may_fail():
+    strat = DatetimeStrategy(dt.datetime.min, dt.datetime.max, none())
+    failure_inducing = find(
+        binary(),
+        lambda b: strat._attempt_one_draw(
+            ConjectureData.for_buffer(b)) is None
+    )
+    # Should raise some kind of error?
```

Comment:
It definitely should. I wonder why it isn't. I think the most likely explanation is that datetimes() is for some reason not exactly the same strategy as strat. Try passing strat instead of datetimes?

The other thing to check if that doesn't work is to make sure it's actually consuming the whole buffer. e.g. do a `data = ConjectureData.for_buffer(failure_inducing)`, pass that to strat._attempt_one_draw,  and assert afterwards that `data.buffer == failure_inducing` and the result